### ISSUE TITLE
react-a11y-anchors should warn when there is no `href` attribute

### ISSUE THREAD

**Issue by astorije** [Microsoft/tslint-microsoft-contrib#434]
At the moment, this plugin warns on `<a href="#">Foobar</a>` but it doesn't warn on `<a>Foobar</a>`. Thoughts on making it warn for this case as well?

For comparison, [`eslint-plugin-jsx-a11y`'s `anchor-is-valid` warns on those](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md):

```tsx
<a />
<a href={undefined} />
<a href={null} />
```

**Comment by sjgllghr** [comment]
If no one else is working on it, I'd like to look into this.

Just to be sure, should this test fail now since we're requiring href, or should it depend on role?
https://github.com/Microsoft/tslint-microsoft-contrib/blob/eac676244291b5dbb8720395d38e2a7929f7e427/src/tests/ReactA11yAnchorsRuleTests.ts#L105-L115
